### PR TITLE
add file upload reuse function for WebGUI apps

### PR DESCRIPTION
**Pull request by heraldantony**
Add reuse function for file upload in WebGUI apps.

### CODE REVIEW

**Review by Templada — DISMISSED** [review]
+1

File: src/reuse/modules/util/file.ts [hunk 1]
```diff
@@ -45,6 +45,43 @@ export class File {
...
+   * @example await util.file.uploadWebGui(["path/to/text1.txt"], "INPUT[title='External file name']");
+   */
+  async uploadWebGui(files: Array<string>, selector: any) {
+    const vl = this.vlf.initLog(this.uploadWebGui);
+    try {
+      if (typeof selector === "string") {
+        const elem = await nonUi5.element.getByCss(selector);
+        await nonUi5.userInteraction.click(elem);
+      } else if (typeof selector === "object") {
+        await ui5.userInteraction.click(selector);
```
Comment (DIFSRIP) [inline src/reuse/modules/util/file.ts:64]: do we need to support the ui5 version in the webgui reuse?

File: src/reuse/modules/util/file.ts [hunk 2]
```diff
@@ -45,6 +45,43 @@ export class File {
...
+      }
+      await common.userInteraction.pressF4();
+      const okButton = await nonUi5.element.getByCss("DIV[id='UpDownDialogChoose']")
+      await nonUi5.assertion.expectToBeVisible(okButton);
+      const fileInput = await nonUi5.element.getByXPath(".//input[@id='webgui_filebrowser_file_upload'][@type='file']", 0, 30000, true);
+      let remoteFiles = "";
+      for (const file of files) {
+        const filePath = this.path.resolve(file);
+        vl.log(`Uploading file with a path ${filePath}`);
```
Comment (DIFSRIP) [inline src/reuse/modules/util/file.ts:73]:
```suggestion
        vl.log(`Uploading file with path ${filePath}`);
```

File: src/reuse/modules/util/file.ts [hunk 3]
```diff
@@ -45,6 +45,43 @@ export class File {
...
+      if (typeof selector === "string") {
+        const elem = await nonUi5.element.getByCss(selector);
+        await nonUi5.userInteraction.click(elem);
+      } else if (typeof selector === "object") {
+        await ui5.userInteraction.click(selector);
+      }
+      await common.userInteraction.pressF4();
+      const okButton = await nonUi5.element.getByCss("DIV[id='UpDownDialogChoose']")
+      await nonUi5.assertion.expectToBeVisible(okButton);
+      const fileInput = await nonUi5.element.getByXPath(".//input[@id='webgui_filebrowser_file_upload'][@type='file']", 0, 30000, true);
```
Comment (DIFSRIP) [inline src/reuse/modules/util/file.ts:69]: please use getByCss, internally the same is called anyways: https://github.com/SAP/wdio-qmate-service/blob/main/src/reuse/modules/nonUi5/element.ts#L269

```suggestion
      const fileInput = await nonUi5.element.getByCss(".//input[@id='webgui_filebrowser_file_upload'][@type='file']", 0, 30000, true);
```

**Review by DIFSRIP — CHANGES_REQUESTED** [review]
please see my comments

File: src/reuse/modules/util/file.ts (hunk 2 above)
Comment (Templada) [reply src/reuse/modules/util/file.ts:73]: agree

File: src/reuse/modules/util/file.ts (hunk 3 above)
Comment (Templada) [reply src/reuse/modules/util/file.ts:69]: also valid

File: src/reuse/modules/util/file.ts (hunk 1 above)
Comment (Templada) [reply src/reuse/modules/util/file.ts:64]: what do you mean by that?

Comment (heraldantony) [reply src/reuse/modules/util/file.ts]: fixed, changed selector type to string. The input element won't be a ui5 element.

File: src/reuse/modules/util/file.ts (hunk 2 above)
Comment (heraldantony) [reply src/reuse/modules/util/file.ts]: fixed :)

File: src/reuse/modules/util/file.ts (hunk 3 above)
Comment (heraldantony) [reply src/reuse/modules/util/file.ts]: changed to getByCss

**Review by DIFSRIP — APPROVED** [review]
looks good

**Pull request merged**